Add Swisscom Internet-Box device tracker

betrisey opened the pull request:
**Description:**
This new device tracker component fetches connected devices from the web interface on the Swisscom Internet-Box
![screenshot web interface](https://cloud.githubusercontent.com/assets/1939395/19833165/be4d2baa-9e38-11e6-9099-5a9118cb8dd6.png)

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#1360

**Example entry for `configuration.yaml` (if applicable):**

``` yaml
device_tracker:
  platform: swisscom
  host: "192.168.1.1" # only required if not 192.168.1.1
```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
- [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
- [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]). (No new dependencies)
- [x] New dependencies are only imported inside functions that use them ([example][ex-import]). (No new dependencies)
- [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`. (No new dependencies)
- [x] New files were added to `.coveragerc`.

betrisey committed:
Add Swisscom Internet-Box device tracker

betrisey committed:
Add Swisscom device tracker to .coveragerc

betrisey committed:
Add timeout to requests
Fix formatting and add missing comments to pass the lint test

balloob commented on homeassistant/components/device_tracker/swisscom.py:
```diff
@@ -0,0 +1,119 @@
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=5)
+
+REQUIREMENTS = ['requests>=2,<3']
```
No need to add requests to the requirements. It's a core dependency of HASS

balloob commented on homeassistant/components/device_tracker/swisscom.py:
```diff
@@ -0,0 +1,119 @@
+
+    def _get_access_token(self):
+        try:
+            r = requests.post('http://' + self.host + '/ws', headers={
```
Please add a timeout.

balloob commented on homeassistant/components/device_tracker/swisscom.py:
```diff
@@ -0,0 +1,119 @@
+
+    def get_swisscom_data(self):
+        """Retrieve data from Swisscom and return parsed result."""
+        r = requests.post('http://' + self.host + '/ws', headers={
```
Add a timeout.

balloob reviewed (changes requested):
Some changes needed and also please have a look at the travis build results. It found quite some linting issues.

betrisey commented:
Hi @balloob 
I've done the changes, now the Travis builds are successful.
Thanks for your help :)

betrisey committed:
Remove authentication which was not required

I realised that there was no need to be authenticated to get the
connected devices. Thanks Swisscom :/

balloob commented on homeassistant/components/device_tracker/swisscom.py:
```diff
@@ -0,0 +1,101 @@
+    def __init__(self, config):
+        """Initialize the scanner."""
+        # If the host is not configured, uses the default IP address.
+        self.host = config.get(CONF_HOST, '192.168.1.1')
```
Can you please move this config format to a `PLATFORM_SCHEMA` and use voluptuous.

betrisey committed:
Moving config to a PLATFORM_SCHEMA and using voluptuous
